datareservoirio/client.py

```diff
@@ -411,6 +414,148 @@ def get(
+            Aggregation period is maximum 24 hours. Values can be in units of h, m, s, ms,
+            microsecond or tick. Use 100 ms instead of 0.1s for 10Hz.
+        max_page_size : optional
+            Maximum number of samples to return per page. The method automatically follows links
+            to next pages and returns the entire series. For advanced usage.
+        Returns
+        -------
+        pandas.Series
+            Series data
+        """
+        if not start:
+            # Required parameter
+            raise ValueError(
+                "You must specify the start date in ISO 8601 format, for example 2023-12-01"
+            )
+
+        if not end:
+            # Required parameter
+            raise ValueError(
+                "You must specify the end date in ISO 8601 format, for example 2023-12-31."
+            )
+
+        if not aggregation_period:
+            # Required parameter
+            raise ValueError(
+                "Aggregation period must be specified using integers and one of these units: h, m, s, ms, microsecond or tick, or their Pandas equivalents"
+            )
+
+        if not aggregation_function:
+            # Required parameter
+            raise ValueError(
+                "Aggregation function must be one of: Avg (mean), Min, Max, Stdev (std)"
+            )
+
+        # Translating some pandas terms to API terms
+        # Note the API is case insensitive so both min and Min will work
+        if aggregation_function in function_translation:
+            aggregation_function = function_translation[aggregation_function]
+
+        for period_unit in period_translation:
+            if (
+                aggregation_period.endswith(period_unit)
+                and aggregation_period[-len(period_unit) - 1].isnumeric()
+            ):
+                aggregation_period = (
+                    aggregation_period[: -len(period_unit)]
+                    + period_translation[period_unit]
+                )
+                break
+
+        start = pd.to_datetime(start, dayfirst=True, unit="ns", utc=True).isoformat()
+        end = pd.to_datetime(end, dayfirst=True, unit="ns", utc=True).isoformat()
+
+        params = {}
+
+        if max_page_size:
+            params["maxPageSize"] = max_page_size
+
+        params["aggregationPeriod"] = aggregation_period
+        params["aggregationFunction"] = aggregation_function
+        params["start"] = start
+        params["end"] = end
+
+        next_page_link = f"{environment.api_base_url}reservoir/timeseries/{series_id}/samples/aggregate?{urlencode(params)}"
+
+        df = (
+            pd.DataFrame(columns=("index", "values"))
+            .astype({"index": "int64"})
+            .astype({"values": "float64"}, errors="ignore")
+        )
+
+        @retry(
+            stop=stop_after_attempt(
+                4
+            ),  # Attempt!, not retry attempt. Attempt 2, is 1 retry
+            retry=retry_if_exception_type(
```

Comment:
In the future, we could, if it becomes an issue, retry using the indications from the server on how long to wait. I am not saying we should do it now, but it is an hook we could look at _if_ this becomes an issue 
<img width="632" alt="image" src="https://github.com/4Subsea/drio-python/assets/88324093/046c1bcd-cb98-49ac-bb4a-df7ac450ff60">

Reply:
Indeed! Let's keep it in mind.